id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java

```diff
@@ -533,11 +535,15 @@ private void updateRequestBodyData(IdRequestDTO request) throws IdRepoAppExcepti
 	@SuppressWarnings({ "rawtypes", "unchecked" })
 	protected void updateVerifiedAttributes(IdRequestDTO<T> requestDTO, DocumentContext inputData, DocumentContext dbData) throws IdRepoAppException {
 		List dbVerifiedAttributes = (List) dbData.read(DOT + VERIFIED_ATTRIBUTES);
+		boolean isV2Version=false;
 		dbVerifiedAttributes.remove(null);
-		boolean isV2Version = requestDTO.getVerifiedAttributes() instanceof Map ? true : false;
+		if (requestDTO.getVerifiedAttributes() != null) {
+			isV2Version = requestDTO.getVerifiedAttributes() instanceof Map;
+		}
 		if (dbVerifiedAttributes.isEmpty()) {
 			dbVerifiedAttributes.add(isV2Version ? new HashMap<>() : new ArrayList<>());
 		}
+		isV2Version = dbVerifiedAttributes.get(0) instanceof Map;
 		if (isV2Version) {
 			Map dbVerifiedAttributeMap = (Map) dbVerifiedAttributes.get(0);
 			Map<String, Object> identityMap = idRepoServiceHelper.convertToMap(requestDTO.getIdentity());
```

Comment:
We should avoid converting identityData to Map more than once. We are invoking this method from the updateIdentity method, where we have already converted it to a Map.

Reply:
removed